javascript/ql/src/semmle/javascript/security/dataflow/Xss.qll

```diff
@@ -302,6 +302,38 @@ module DomBasedXss {
   private class MetacharEscapeSanitizer extends Sanitizer, Shared::MetacharEscapeSanitizer { }
 
   private class UriEncodingSanitizer extends Sanitizer, Shared::UriEncodingSanitizer { }
+
+  /**
+   * Holds if there exists two dataflow edges to `succ`, where one edges is sanitized, and the other edge starts with `pred`. 
+   */
+  predicate isOptionallySanitizedEdge(DataFlow::Node pred, DataFlow::Node succ) {
+    exists(DataFlow::CallNode sanitizer |
+      sanitizer.getCalleeName().regexpMatch("(?i).*sanitize.*")
```

Comment:
Could you use `HtmlSanitizerCall` instead, and extend it to ensure `DOMPurify.sanitize` is matched?

Reply:
One of the CVEs use their own sanitizer, so we would not recognize that if we use `HtmlSanitizerCall`.  

How about I add `or sanitizer instanceof HtmlSanitizerCall`?

Reply:
Would the name heuristic in `HtmlSanitzerCall` match that?

Reply:
> Would the name heuristic in `HtmlSanitzerCall` match that?

It does, I missed that when I looked at `HtmlSanitzerCall` the first time.  
I implemented your suggestion. We still get TN on the 2 CVEs.